apps/ejabberd/src/ejabberd_admin.erl

```diff
@@ -300,16 +300,16 @@ send_service_message_all_mucs(Subject, AnnouncementText) ->
                                       | {'ok', io_lib:chars()}.
 register(User, Host, Password) ->
     case ejabberd_auth:try_register(User, Host, Password) of
-        ok ->
-            {ok, io_lib:format("User ~s@~s successfully registered", [User, Host])};
         {error, exists} ->
             String = io_lib:format("User ~s@~s already registered at node ~p",
                                    [User, Host, node()]),
             {exists, String};
         {error, Reason} ->
             String = io_lib:format("Can't register user ~s@~s at node ~p: ~p",
                                    [User, Host, node(), Reason]),
-            {cannot_register, String}
+            {cannot_register, String};
+        _ ->
```

Comment:
I understand the response for successful registration changed. Can we have the new value here instead of accepting anything? If I understand it correctly, the new result is some empty `mongoose_stanza` accumulator. If so I insist on rewriting the hooks a bit and returning {ok, Acc} in case of successful hook's processing in a handler.